### PR TITLE
[docs]fix typo in substring docs

### PR DESCRIPTION
**Pull request by cambyzju**
# Proposed changes

Issue Number: close #xxx

## Problem Summary:

Describe the overview of changes.
1. fix typo in docs/en/docs/sql-manual/sql-functions/string-functions/substring.md
2. add another substring example with pos=0;

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
4. Has unit tests been added: (Yes/No/No Need)
5. Has document been added or modified: (Yes/No/No Need)
6. Does it need to update dependencies: (Yes/No)
7. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: docs/en/docs/sql-manual/sql-functions/string-functions/substring.md [hunk 1]
```diff
@@ -74,4 +82,4 @@ mysql> select substring('abc1def', 2, 2);
 ```
 
 ### keywords
-SUBSTRING
+SUBSTRING,STRING
```
Comment (jackwener) [inline docs/en/docs/sql-manual/sql-functions/string-functions/substring.md:85]:
```suggestion
SUBSTRING, STRING
```

File: docs/zh-CN/docs/sql-manual/sql-functions/string-functions/substring.md [hunk 2]
```diff
@@ -69,4 +76,4 @@ mysql> select substring('abc1def', 2, 2);
 +-----------------------------+
 ```
 ### keywords
-SUBSTRING
+SUBSTRING,STRING
```
Comment (jackwener) [inline docs/zh-CN/docs/sql-manual/sql-functions/string-functions/substring.md:79]:
```suggestion
SUBSTRING, STRING
```

File: docs/en/docs/sql-manual/sql-functions/string-functions/substring.md (hunk 1 above)
Comment (cambyzju) [reply docs/en/docs/sql-manual/sql-functions/string-functions/substring.md]: done

File: docs/zh-CN/docs/sql-manual/sql-functions/string-functions/substring.md (hunk 2 above)
Comment (cambyzju) [reply docs/zh-CN/docs/sql-manual/sql-functions/string-functions/substring.md]: done

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**